frontend/src/utils/api/response.ts

```diff
@@ -0,0 +1,76 @@
+import { ProviderContext, VariantType } from 'notistack';
+import { ApiError } from 'src/services/openapi';
+import {
+  contactAdministrator,
+  showErrorAlert,
+  showInfoAlert,
+} from 'src/utils/notifications';
+
+/**
+ * Display all errors contained in an `ApiError` object.
+ *
+ * This function provides a generic way to display all error messages returned
+ * by an API call. It can be used in every form to harmonize the messages
+ * processing.
+ *
+ * Ex
+ *      const response = await addNewVideo({ video_id }).catch(
+ *        (reason: ApiError) => {
+ *          displayErrors(enqueueSnackbar, reason);
+ *        }
+ *      );
+ *
+ * The optional `overrideStatuses` parameter allows the developers to display
+ * in the UI custom messages for a specific set of HTTP response status code.
+ * They will override those contained in the response.
+ *
+ * Ex
+ *      const response = await addNewVideo({ video_id }).catch(
+ *        (reason: ApiError) => {
+ *          displayErrors(
+ *            enqueueSnackbar,
+ *            reason,
+ *            'Sorry, an error has occurred. The video cannot be added.',
+ *            [{ status: 409, variant: 'error', msg: 'Video already added.' }]
+ *          );
+ *        }
+ *      );
+ *
+ * Unknown `Error` or malformed `ApiError` are displayed as an invitation to
+ * try again later or contact an administrator.
+ */
+export const displayErrors = (
+  display: ProviderContext['enqueueSnackbar'],
+  reason: ApiError | Error,
+  message?: string,
+  overrideStatuses?: Array<{
+    status: number;
+    msg: string;
+    variant: VariantType;
+  }>
+) => {
+  if (reason && 'body' in reason) {
+    // override message from the API
+    if (overrideStatuses) {
+      const status = overrideStatuses.filter(
+        (elem) => elem.status === reason.status
+      );
+      if (status) {
+        display(status[0].msg, { variant: status[0].variant });
+        return;
+      }
+    }
+
+    // HTTP 429 Too Many Requests are considered information
+    if (reason.status === 429) {
+      showInfoAlert(display, reason.body['detail']);
+    } else {
+      const newErrorMessages = Object.values(reason['body']).flat();
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      newErrorMessages.map((msg) => showErrorAlert(display, msg));
```

Comment:
It's always possible to use `as` to force a specific type when it's "unknown" for Typescript.  

And we have no guarantee about the content of `reason.body` here so it will be hard to be completely safe here.  
A bug had already been reported when the body is a raw string:
![image](https://user-images.githubusercontent.com/4726554/148229567-7a74b5a8-da32-4c09-a42a-11361f54fea4.png)
```
Due to the behavior of `Object.values()`:
> Object.values("Error")
Array(5) [ "E", "r", "r", "o", "r" ]
```

A slightly safer version would be:
```suggestion
      const body = reason.body;
      const newErrorMessages =
        typeof body === 'string'
          ? [body]
          : Object.values(reason['body']).flat();
      newErrorMessages.map((msg) => showErrorAlert(display, msg as string));
```